packages/front-end/components/Features/FeatureModal.tsx

```diff
@@ -61,62 +62,88 @@ function parseDefaultValue(
   }
 }
 
+const initEnvironmentSettings = ({
+  environments,
+  featureToDuplicate,
+  permissions,
+  project,
+}: {
+  environments: ReturnType<typeof useEnvironments>;
+  featureToDuplicate?: FeatureInterface;
+  permissions: ReturnType<typeof usePermissions>;
+  project: string;
+}): Record<string, FeatureEnvironment> => {
+  const envSettings: Record<string, FeatureEnvironment> = {};
+
+  environments.forEach((e) => {
+    const canPublish = permissions.check("publishFeatures", project, [e.id]);
+    const defaultEnabled = canPublish ? e.defaultState ?? true : false;
+    const enabled = canPublish
+      ? featureToDuplicate?.environmentSettings?.[e.id]?.enabled ??
+        defaultEnabled
+      : false;
+    const rules = featureToDuplicate?.environmentSettings?.[e.id]?.rules ?? [];
+
+    envSettings[e.id] = { enabled, rules };
+  });
+
+  return envSettings;
+};
+
 export default function FeatureModal({
   close,
   onSuccess,
   inline,
   cta = "Create",
   secondaryCTA,
+  featureToDuplicate,
 }: Props) {
   const { project, refreshTags } = useDefinitions();
   const environments = useEnvironments();
   const permissions = usePermissions();
-
   const { refreshWatching } = useWatching();
 
-  const defaultEnvSettings: Record<string, FeatureEnvironment> = {};
-  environments.forEach((e) => {
-    let enabled = e.defaultState ?? true;
-    if (!permissions.check("publishFeatures", project, [e.id])) enabled = false;
-
-    defaultEnvSettings[e.id] = {
-      enabled,
-      rules: [],
-    };
+  const initialEnvSettings = initEnvironmentSettings({
+    environments,
+    featureToDuplicate,
+    permissions,
+    project,
   });
 
   const form = useForm({
     defaultValues: {
-      valueType: "boolean",
-      defaultValue: getDefaultValue("boolean"),
+      valueType: featureToDuplicate ? featureToDuplicate.valueType : "boolean",
+      defaultValue: featureToDuplicate
+        ? featureToDuplicate.defaultValue
+        : getDefaultValue("boolean"),
       description: "",
       id: "",
```

Comment:
I might suggest carrying the id over as well, but appending `" (duplicate)"` or similar

Reply:
In the form provided, the `id` is the one property editable. It is more convenient for the user to start from a blank field than to delete the value and retype it, so I would disagree with prepopulating the `id` and appending `duplicate`. The feature `id` needs to be alphanumeric snake case so the suggested prefilled value wouldn't be valid.

Reply:
I ended up implementing a rudimentary feature key generation algorithm for duplicated features (since we have the entire list of features available to us on the frontend)